nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/csv/DocumentModelCSVWriter.java

```diff
@@ -18,120 +18,185 @@
 import org.nuxeo.ecm.core.io.registry.Writer;
 import org.nuxeo.ecm.core.io.registry.reflect.Setup;
-import org.nuxeo.ecm.core.schema.SchemaManager;
 import org.nuxeo.ecm.core.schema.types.Field;
 import org.nuxeo.ecm.core.schema.types.Schema;
-import org.nuxeo.ecm.core.schema.utils.DateParser;
 
 /**
  * @since 10.3
  */
 @Setup(mode = SINGLETON, priority = REFERENCE)
 public class DocumentModelCSVWriter extends AbstractCSVWriter<DocumentModel> {
 
+    public static final String SCHEMAS_CTX_DATA = "schemas";
+
+    public static final String XPATHS_CTX_DATA = "xpaths";
+
     public DocumentModelCSVWriter() {
-        super(DocumentModel.class);
+        super();
     }
 
-    @Inject
-    protected SchemaManager schemaManager;
-
-    @Inject
-    protected MarshallerRegistry registry;
+    @Override
+    protected void write(DocumentModel entity, CSVPrinter printer) throws IOException {
+        writeSystem(entity, printer);
+        // If no specific schema is defined, all schemas are written
+        List<String> schemas = ctx.getParameter(SCHEMAS_CTX_DATA) != null ? ctx.getParameter(SCHEMAS_CTX_DATA)
+                : Arrays.asList(entity.getSchemas());
+        List<String> xpaths = ctx.getParameter(XPATHS_CTX_DATA);
+        Collections.sort(schemas);
+        for (String schema : schemas) {
+            writeSchema(entity, schema, printer);
+        }
+        if (xpaths != null) {
+            Collections.sort(xpaths);
+            for (String xpath : xpaths) {
+                writeProperty(entity, xpath, printer);
+            }
+        }
+    }
 
     @Override
-    public void write(DocumentModel entity, Class<?> clazz, Type genericType, MediaType mediatype, OutputStream out)
-            throws IOException {
-        writeSystem(out, entity);
-        // schemas are sorted for reproducibility
-        List<String> schemas = new ArrayList<>(Arrays.asList(entity.getSchemas()));
+    protected void writeHeader(DocumentModel entity, CSVPrinter printer) throws IOException {
+        writeDocumentModelHeader(entity, printer);
+        printer.println();
+    }
+
+    public void writeDocumentModelHeader(DocumentModel docModel, CSVPrinter printer) throws IOException {
+        printer.print(REPOSITORY_FIELD);
+        printer.print(UID_FIELD);
+        printer.print(PATH_FIELD);
+        printer.print(TYPE_FIELD);
+        printer.print(STATE_FIELD);
+        printer.print(PARENT_REF_FIELD);
+        printer.print(IS_CHECKED_OUT_FIELD);
+        printer.print(IS_VERSION_FIELD);
+        printer.print(IS_PROXY_FIELD);
+        printer.print(PROXY_TARGET_ID_FIELD);
+        printer.print(VERSIONABLE_ID_FIELD);
+        printer.print(CHANGE_TOKEN_FIELD);
+        printer.print(IS_TRASHED_FIELD);
+        printer.print(TITLE_FIELD);
+        printer.print(VERSION_LABEL_FIELD);
+        printer.print(LOCK_OWNER_FIELD);
+        printer.print(LOCK_CREATED_FIELD);
+        printer.print(LAST_MODIFIED_FIELD);
```

Comment:
You could put them all in a static list and do `THE_LIST.forEach(printer::print)`